### ISSUE TITLE
[[nodiscard]] on numeric_limits functions

### ISSUE THREAD

**Issue by beached** [llvm/llvm-project#83695]
Would it be welcome to others to add `[[nodiscard]]` to the static methods on `numeric_limits`.   It would help point out bugs like 

```cpp
if( lhs == std::numeric_limits<T>::min( ) and rhs == T{ -1 } ) {
  std::numeric_limits<T>::max( ); /* oops, ignored it */
}
```

**Comment by Sirraide** [comment]
The standard doesn’t specify that they’re `[[nodiscard]]`, so I’m not sure we can just do that...

**Comment by beached** [comment]
I thought implementors had freedom here, I can try and find it as it's not in the obvious place `[dcl.attr.nodiscard]`

**Comment by MagentaTreehouse** [comment]
The [MSVC STL implementation](https://github.com/microsoft/STL/blob/main/stl/inc/limits) has `[[nodiscard]]`.

**Comment by beached** [comment]
I guess the issue is that 

```cpp
#include <limits>

int main() {
    std::numeric_limits<int>::max( );
}
``` 
doesn't warn.  `-Wunused` But 

```cpp
#include <limits>

int main( ) {
  5;  // warns
  auto a = std::numeric_limits<int>::max( ); // warns
}
```
does warn.

**Comment by philnik777** [comment]
It's absolutely fine to add `[[nodiscard]]` in places where the standard doesn't add it. Its only job is it to make the compiler produce a warning, which the standard has no constraints or requirements on. Libc++ has `_LIBCPP_NODISCARD_EXT` for cases which aren't in the standard.

**Comment by Sirraide** [comment]
Ah yeah, that’s true; I guess I just wasn’t sure because it’s a standard attribute, but it also makes sense that it would be fine.

**Comment by mordante** [comment]
[P3122R0](https://wg21.link/p3122r0) "[[nodiscard]] should be Recommended Practice" suggests to make these things recommended practice. (The author of the paper is the maintainer of libstdc++.) If that gets accepted we probably want to consider adding `[[nodiscard]]` in more places unconditionally and even removing `_LIBCPP_NODISCARD_EXT`.
